### PR TITLE
OwnedFeeCollector for Arbitrum perps IFeeCollector - audit modifications 

### PR DESCRIPTION
**Pull request by fredsnax**
…0) in constructor and slight gas improvement in claimFees removing modifier

### CODE REVIEW

File: auxiliary/OwnedFeeCollector/test/integration/OwnedFeeCollector.test.ts [hunk 1]
```diff
@@ -6,13 +6,14 @@ import { ethers } from 'ethers';
...
-  const { getContract, owner, user } = bootstrapOwnedFeeCollector();
+  const { getContract, owner, user, feeShareController } = bootstrapOwnedFeeCollector();
 
   let OwnedFeeCollector: ethers.Contract;
   let UsdToken: ethers.Contract;
 
   const usdAmount = bn(5000);
-  const ownerFeeShareRatio = bn(0.5);
+  const feeShareRatio = bn(0.5);
+  const zeroAddress = '0x0000000000000000000000000000000000000000';
```
Comment (leomassazza) [inline auxiliary/OwnedFeeCollector/test/integration/OwnedFeeCollector.test.ts:16]: you can use `ethers.constants.AddressZero`

**Comment by fredsnax** [comment]
Changes in this PR: 

- according to the audit notes using this error pattern instead of the modifiers is slightly more gas efficient and we already use the error pattern throughout so I changed it although its not that important imo

- audit suggestion to add the ability to change the fee share. at first I thought it was better immutable but this is actually important since we are changing the fee share for the first month of arbitrum perps. Kaleb agreed we should be able to change it in the contract rather than deploy a new one.

- ability to change the fee share controller address with this new functionality of changing the fee share. think the pDAO controls the fee share and the TC claims the fees

- adds zero address check in the controller

**Comment by fredsnax** [comment]
further changes after review by Leo: pdao is the owner. tc is the fee share recipient and can claim fees. pdao handles all other admin and can also claim fees to the fee share recipient or change the recipient address

File: auxiliary/OwnedFeeCollector/cannonfile.toml [hunk 2]
```diff
@@ -3,9 +3,12 @@ version = "<%= package.version %>"
...
+defaultValue = "0xBE0eB53F46cd790Cd13851d5EFf43D12404d33E8"
+
+[setting.fee_share_recipient]
 defaultValue = "0x0000000000000000000000000000000000000000"
 
-[setting.owner_fee_share]
+[setting.fee_share]
 defaultValue = "500000000000000000"
 
 [setting.fee_token]
```
Comment (barrasso) [inline auxiliary/OwnedFeeCollector/cannonfile.toml:14]: `fee_token` should be nonzero as it is set in the constructor

**Comment by barrasso** [comment]
thanks fred, will take this over from here 🫡

File: auxiliary/OwnedFeeCollector/cannonfile.toml [hunk 3]
```diff
@@ -3,9 +3,12 @@ version = "<%= package.version %>"
 description = "Colelcts fees on behalf of an owner"
 
 [setting.owner]
+defaultValue = "0xBE0eB53F46cd790Cd13851d5EFf43D12404d33E8"
+
+[setting.fee_share_recipient]
 defaultValue = "0x0000000000000000000000000000000000000000"
```
Comment (barrasso) [inline auxiliary/OwnedFeeCollector/cannonfile.toml:9]: `fee_share_recipient` cannot b 0

**Review by barrasso — APPROVED** [review]

**Pull request merged**